server.go

```diff
@@ -333,6 +339,16 @@ func (sp *serverPeer) OnVersion(p *peer.Peer, msg *wire.MsgVersion) *wire.MsgRej
 		return nil
 	}
 
+	// Reject outbound peers that are not full nodes.
+	wantServices := wire.SFNodeNetwork
+	if !sp.Inbound() && !hasServices(msg.Services, wantServices) {
+		srvrLog.Debugf("Rejecting peer %s with services %v due to not "+
+			"providing desired services %v", msg.Services, wantServices)
```

Comment:
We could improve this by masking everything off except the missing services: `wantServices&^msg.Services`

Reply:
Doesn't really matter for dcrd's case as it only needs a single service flag, so it would be the same result either way, but that could change in the future.

Reply:
Agreed, that currently it's the same, but I think it's a good idea, because I would expect there to be pruned nodes in the future with a new service bit, in which case it would matter.

Reply:
Updated.